### ISSUE TITLE
When training Qwen2.5VL using recipe/entropy, the grad_norm and pg_loss are extremely large, exceeding e10.

### ISSUE THREAD

**Issue by YankaiChen0308** [volcengine/verl#2721]
When training Qwen2.5VL using recipe/entropy, the grad_norm and pg_loss are extremely large, exceeding e10. Below is my configuration. What could be the potential issues? (Some entries have been redacted due to privacy concerns)
`#!/usr/bin/env bash
set -xeuo pipefail

project_name='Entropy'
exp_name='klcov-Qwen2.5vl-3B'

adv_estimator=grpo

use_kl_in_reward=False
kl_coef=0.0
use_kl_loss=False
kl_loss_coef=0.0

clip_ratio_low=0.2
clip_ratio_high=0.2

max_prompt_length=$((1024))
max_response_length=$((1024))
enable_overlong_buffer=False
overlong_buffer_len=$((1024))
overlong_penalty_factor=1.0

loss_agg_mode="token-mean"
loss_mode="kl_cov"
enable_filter_groups=True
filter_groups_metric=score_bool
max_num_gen_batches=4
train_prompt_bsz=16
gen_prompt_bsz=$((train_prompt_bsz * 3))
train_prompt_mini_bsz=8
n_resp_per_prompt=8
actor_ppo_max_token_len=$((max_prompt_length + max_response_length))
infer_ppo_max_token_len=$((max_prompt_length + max_response_length))
vllm_dtype=float16
#max_token=16384

# Ray
RAY_ADDRESS=${RAY_ADDRESS:-"http://localhost:8265"}
NNODES=${NNODES:-1}
# Paths
RAY_DATA_HOME=""
MODEL_PATH="/mnt/models/Qwen2.5-VL-3B-Instruct"
CKPTS_DIR=${CKPTS_DIR:-"${RAY_DATA_HOME}/checkpoints/${project_name}/${exp_name}"}
TRAIN_FILE="" 
TEST_FILE=""

# Algorithm
temperature=1.0
top_p=1.0
top_k=-1 # 0 for HF rollout, -1 for vLLM rollout
ppo_kl_coef=1
kl_cov_ratio=0.002
lr=5e-7

# Mathematically equivalent
use_dynamic_bsz=False
act_micro_bsz=1
ref_rollout_micro_bsz=4
offload=True
gen_tp=4
enable_chunked_prefill=False

# reward func use data_source to choose
custom_reward_function_path="xxxxx"
custom_reward_function_name="xxxxx"

CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 python3 -m recipe.entropy.main_entropy \
    data.train_files="${TRAIN_FILE}" \
    data.val_files="${TEST_FILE}" \
    data.prompt_key=prompt \
    data.truncation='error' \
    data.filter_overlong_prompts=True \
    data.max_prompt_length=${max_prompt_length} \
    data.max_response_length=${max_response_length} \
    data.gen_batch_size=${gen_prompt_bsz} \
    data.train_batch_size=${train_prompt_bsz} \
    data.return_raw_chat=True \
    data.validation_shuffle=True \
    actor_rollout_ref.rollout.n=${n_resp_per_prompt} \
    actor_rollout_ref.actor.use_kl_loss=${use_kl_loss} \
    actor_rollout_ref.actor.kl_loss_coef=${kl_loss_coef} \
    actor_rollout_ref.actor.clip_ratio_low=${clip_ratio_low} \
    actor_rollout_ref.actor.clip_ratio_high=${clip_ratio_high} \
    actor_rollout_ref.actor.clip_ratio_c=10.0 \
    actor_rollout_ref.actor.policy_loss.loss_mode=${loss_mode} \
    actor_rollout_ref.actor.policy_loss.kl_cov_ratio=${kl_cov_ratio} \
    actor_rollout_ref.actor.policy_loss.ppo_kl_coef=${ppo_kl_coef} \
    actor_rollout_ref.actor.ppo_micro_batch_size_per_gpu=${act_micro_bsz} \
    actor_rollout_ref.actor.entropy_from_logits_with_chunking=True \
    actor_rollout_ref.actor.fsdp_config.reshard_after_forward=False \
    actor_rollout_ref.ref.entropy_from_logits_with_chunking=True \
    actor_rollout_ref.ref.fsdp_config.forward_prefetch=True \
    actor_rollout_ref.ref.fsdp_config.reshard_after_forward=False \
    actor_rollout_ref.actor.fsdp_config.forward_prefetch=True \
    actor_rollout_ref.rollout.mode=sync \
    actor_rollout_ref.rollout.dtype=${vllm_dtype} \
    actor_rollout_ref.rollout.engine_kwargs.vllm.disable_mm_preprocessor_cache=False \
    actor_rollout_ref.ref.fsdp_config.forward_prefetch=True \
    actor_rollout_ref.actor.entropy_checkpointing=True \
    actor_rollout_ref.ref.entropy_checkpointing=True \
    actor_rollout_ref.ref.entropy_from_logits_with_chunking=True \
    custom_reward_function.path=${custom_reward_function_path} \
    custom_reward_function.name=${custom_reward_function_name} \
    algorithm.adv_estimator=${adv_estimator} \
    algorithm.use_kl_in_reward=${use_kl_in_reward} \
    algorithm.kl_ctrl.kl_coef=${kl_coef} \
    algorithm.filter_groups.enable=${enable_filter_groups} \
    algorithm.filter_groups.metric=${filter_groups_metric} \
    algorithm.filter_groups.max_num_gen_batches=${max_num_gen_batches} \
    actor_rollout_ref.model.use_remove_padding=True \
    actor_rollout_ref.model.enable_gradient_checkpointing=True \
    actor_rollout_ref.model.use_shm=True \
    actor_rollout_ref.model.enable_activation_offload=True \
    actor_rollout_ref.actor.use_dynamic_bsz=${use_dynamic_bsz} \
    actor_rollout_ref.ref.log_prob_use_dynamic_bsz=${use_dynamic_bsz} \
    actor_rollout_ref.rollout.log_prob_use_dynamic_bsz=${use_dynamic_bsz} \
    actor_rollout_ref.rollout.log_prob_micro_batch_size_per_gpu=${ref_rollout_micro_bsz} \
    actor_rollout_ref.ref.log_prob_micro_batch_size_per_gpu=${ref_rollout_micro_bsz} \
    actor_rollout_ref.model.path="${MODEL_PATH}" \
    actor_rollout_ref.model.enable_gradient_checkpointing=True \
    actor_rollout_ref.actor.optim.lr=1e-6 \
    actor_rollout_ref.actor.optim.weight_decay=0.1 \
    actor_rollout_ref.actor.optim.lr_warmup_steps=10 \
    actor_rollout_ref.actor.ppo_mini_batch_size=${train_prompt_mini_bsz} \
    actor_rollout_ref.actor.fsdp_config.param_offload=${offload} \
    actor_rollout_ref.actor.fsdp_config.optimizer_offload=${offload} \
    actor_rollout_ref.actor.entropy_coeff=0 \
    actor_rollout_ref.actor.grad_clip=1.0 \
    actor_rollout_ref.actor.loss_agg_mode=${loss_agg_mode} \
    actor_rollout_ref.actor.ulysses_sequence_parallel_size=1 \
    actor_rollout_ref.rollout.gpu_memory_utilization=0.6 \
    actor_rollout_ref.rollout.tensor_model_parallel_size=4 \
    actor_rollout_ref.rollout.enable_chunked_prefill=${enable_chunked_prefill} \
    actor_rollout_ref.rollout.temperature=${temperature} \
    actor_rollout_ref.rollout.top_p=${top_p} \
    actor_rollout_ref.rollout.top_k="${top_k}" \
    actor_rollout_ref.rollout.val_kwargs.temperature=${temperature} \
    actor_rollout_ref.rollout.val_kwargs.top_p=${top_p} \
    actor_rollout_ref.rollout.val_kwargs.top_k=${top_k} \
    actor_rollout_ref.rollout.val_kwargs.do_sample=False \
    actor_rollout_ref.rollout.val_kwargs.n=1 \
    actor_rollout_ref.ref.fsdp_config.param_offload=${offload} \
    actor_rollout_ref.ref.ulysses_sequence_parallel_size=1 \
    actor_rollout_ref.actor.fsdp_config.fsdp_size=-1 \
    reward_model.reward_manager=dapo \
    reward_model.overlong_buffer.enable=${enable_overlong_buffer} \
    reward_model.overlong_buffer.len=${overlong_buffer_len} \
    reward_model.overlong_buffer.penalty_factor=${overlong_penalty_factor} \
    trainer.logger='["console","tensorboard"]' \
    trainer.project_name="${project_name}" \
    trainer.experiment_name="${exp_name}" \
    trainer.n_gpus_per_node=8 \
    trainer.nnodes="${NNODES}" \
    trainer.val_before_train=False \
    trainer.test_freq=5 \
    trainer.save_freq=20 \
    trainer.total_epochs=200 \
    trainer.default_local_dir="${CKPTS_DIR}" \
    trainer.resume_mode=auto
`

**Comment by lwp-hhxx** [comment]
I have the same problem, qwen2.5-7b-math, grad_norm and pg_loss are very large

**Comment by Clementine24** [comment]
same issue, any update?

**Comment by Kirrito-k423** [comment]
### 中文
定时清理检查发现：此 issue 最近一次非当前运行账号的公开回复或引用时间是 `2025-09-26T13:07:34Z`（comment，来自 `Clementine24`），相对于本轮运行日期 `2026-07-12` 已超过 6 个月。因此按社区 issue 定时清理标准关闭。

已检查证据：Latest non-self public reply/reference: `2025-09-26T13:07:34Z` (comment by `Clementine24`).
Comments API: https://api.github.com/repos/verl-project/verl/issues/2721/comments
Timeline API: https://api.github.com/repos/verl-project/verl/issues/2721/timeline

如果作者或社区成员仍能提供新的复现、证据或需求背景，请在此 issue 下继续评论请求重开，或创建新的 issue 并链接到这里。

### English
Scheduled cleanup check found that the latest public reply or reference by an account other than the current automation account was `2025-09-26T13:07:34Z` (comment by `Clementine24`), which is more than 6 months before this run date `2026-07-12`. Therefore this issue is being closed under the scheduled stale-issue cleanup rule.

Evidence checked: Latest non-self public reply/reference: `2025-09-26T13:07:34Z` (comment by `Clementine24`).
Comments API: https://api.github.com/repos/verl-project/verl/issues/2721/comments
Timeline API: https://api.github.com/repos/verl-project/verl/issues/2721/timeline

If the author or community members can still provide a fresh reproduction, evidence, or requirement context, please comment here to request reopening, or open a new issue and link back to this one.

This response was generated by a scheduled Codex GPT5.5 issue processing task.
